src/main/java/com/passpnu/passwordmanager/service/PasswordEntityServiceImpl.java

```diff
@@ -171,4 +212,65 @@ public void deletePassword(Long serviceId, AuthUserDetailsDto userDto) {
         Long userId = userDto.getId();
         passwordRepository.deleteByUserIdAndServiceId(userId, serviceId);
     }
+
+    @Override
+    public List<AnalysisDto> analyzeSystem(AuthUserDetailsDto authUser) throws EncryptionException, NameNotFoundException {
+        final String encryptionKey = authUser.getEncryptionKey();
+
+        List<PasswordEntity> entities = passwordRepository.findAllByUserId(authUser.getId());
+
+        List<AnalysisDto> analysisDtoList = entities.stream()
+                .map(entity -> {
+                    try {
+                        return customPasswordMapper.entityToAnalysisDto(entity, passwordEncryptor, encryptionKey);
+                    } catch (EncryptionException e) {
+                        throw new UncheckedEncryptionException(e.getMessage());
+                    }
+                })
+                .toList();
+
+        final int listSize = analysisDtoList.size();
+        String firstPassword;
+        String secondPassword;
+        AnalysisDto firstAnalysisDto;
+        AnalysisDto secondAnalysisDto;
+        double similarity;
+        SimilarPasswordDto firstToSecond;
+        SimilarPasswordDto secondToFirst;
+
+        for(int i = 0; i < listSize; i++){
+            firstAnalysisDto = analysisDtoList.get(i);
+            firstPassword = firstAnalysisDto.getPassword();
+            firstAnalysisDto.setStrong(passwordUtil.isPasswordStrong(firstPassword).isStrong());
```

Comment:
There can be two separate methods: for checking strength of every password and for checking similarity

Reply:
fixed